safenode/src/domain/storage/spends.rs

```diff
@@ -199,6 +182,34 @@ impl SpendStorage {
         Ok(())
     }
 
+    async fn add(&mut self, signed_spend: &SignedSpend) -> Result<()> {
+        let addr = dbc_address(signed_spend.dbc_id());
+        let filepath = self.address_to_filepath(&addr, &self.valid_spends_path)?;
+
+        if filepath.exists() {
+            self.validate(signed_spend).await?;
+            return Ok(());
+        }
+
+        // Store the spend to local file system.
```

Comment:
does it need to be stored to network ?

Reply:
This is the node storage in the network. So, by storing it here, the spend is stored in the network (and considered so by the world when majority of close group has it).